src/analyses/goto_check.cpp

```diff
@@ -2453,3 +2492,48 @@ void goto_check(
   const namespacet ns(goto_model.symbol_table);
   goto_check(ns, options, goto_model.goto_functions, message_handler);
 }
+
+void goto_checkt::add_active_named_check_pragmas(
+  source_locationt &source_location)
+{
+  for(auto it : name_to_flag)
+    if(*(it.second))
+      source_location.add_pragma("checked:" + id2string(it.first));
+}
+
+void goto_checkt::add_all_disable_named_check_pragmas(
+  source_locationt &source_location)
+{
+  for(auto it : name_to_flag)
+    source_location.add_pragma("disable:" + id2string(it.first));
+}
+
+goto_checkt::named_check_statust
+goto_checkt::match_named_check(irep_idt named_check)
+{
+  auto s = id2string(named_check);
+  auto status = check_statust::DISABLE;
+  if(s.find("enable:") == 0)
+  {
+    status = check_statust::ENABLE;
+  }
+  else if(s.find("disable:") == 0)
+  {
+    status = check_statust::DISABLE;
+  }
+  else if(s.find("checked:") == 0)
+  {
+    status = check_statust::CHECKED;
+  }
+  else
+  {
+    return {};
+  }
+
+  auto sep = s.find(":");
```

Comment:
Each of the above branches would actually have a constant known value for this. Perhaps avoid the duplicate `find`? Alternatively, do the `find` early on and then just compare the substrings up to that known position of the colon.